### PR TITLE
WGSL implement texture intrinsics except gather and sampler-less

### PR DESCRIPTION
**Pull request by jkwak-work**
This PR implements all of the texture intrinsics for WGSL except "Gather" and sampler-less.
They will be implemented in a separate PR.

A few things to note:
- texture sampling functions are available only for the fragment shader stage; not for compute
- WGSL doesn't have any functions similar to CalculateLevelOfDetail or CalculateLevelOfDetailUnclamped.
- WGSL doesn't have a function overlaoding for `textureSample` with "clamp" or "status" arguments.
- WGSL doesn't support Load operation with offset for texture_multisampled_XX and texture_storage_XX.
- WGSL supports only four types of depth textures: 2D, 2D_array, cube and cube_array.
- WGSL doesn't support "offset" variants for cube and cube_array.

### CODE REVIEW

**Comment by jkwak-work** [comment]
It seems that debug builds are failing while compiling stdlib.
I am investigating.

File: tests/wgsl/texture.slang [hunk 1]
```diff
@@ -0,0 +1,657 @@
...
+TextureCube<uint16_t4> tCube_u16;
+
+//TEST_INPUT: Texture1D(size=4, content = zero, arrayLength=2):name t1DArray_u16
+Texture1DArray<uint16_t4> t1DArray_u16;
+//TEST_INPUT: Texture2D(size=4, content = zero, arrayLength=2):name t2DArray_u16
+Texture2DArray<uint16_t4> t2DArray_u16;
+//TEST_INPUT: TextureCube(size=4, content = zero, arrayLength=2):name tCubeArray_u16
+TextureCubeArray<uint16_t4> tCubeArray_u16;
+
+// Metal doc says "For depth texture types, T must be float."
```
Comment (aleino-nv) [inline tests/wgsl/texture.slang]: Metal?

File: tests/wgsl/texture.slang [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+    Texture1D<vector<T,N>> t1D,
+    Texture2D<vector<T,N>> t2D,
+    Texture3D<vector<T,N>> t3D,
+    TextureCube<vector<T,N>> tCube,
+    Texture1DArray<vector<T,N>> t1DArray,
+    Texture2DArray<vector<T,N>> t2DArray,
+    TextureCubeArray<vector<T,N>> tCubeArray
+)
+{
+    // METAL-LABEL: TEST_texture
```
Comment (aleino-nv) [inline tests/wgsl/texture.slang]: Metal again, and in some other places below.
Why does it say "metal" everywhere?

Comment (aleino-nv) [reply tests/wgsl/texture.slang]: Oh I see it's fixed in a later commit.

**Review by aleino-nv — APPROVED** [review]
LGTM

**Comment by aleino-nv** [comment]
> LGTM

(Assuming the CI issue is resolved.)

File: tests/wgsl/texture.slang (hunk 2 above)
Comment (jkwak-work) [reply tests/wgsl/texture.slang]: Sorry about the confusion.
The test file is copied from tests/metal/texture.slang and it was incrementally modified/fixed as I work through the implementation.
The keyword, "Metal", is no longer used in the test file at the latest commit.

**Comment by jkwak-work** [comment]
I think I fixed the CI failure.
There was a function that could have no `return` or `__intrinsic_asm` statement.

File: tests/wgsl/texture.slang [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+    uint numLevels = 0, elements = 0, sampleCount = 0;
+    float fnumLevels = 0.0f, felements = 0.0f;
+
+    bool voidResult = true;
+
+    // ======================
+    //  void GetDimensions()
+    // ======================
+
+    // WGSL doesn't support the mip level query;
```
Comment (jkwak-work) [inline tests/wgsl/texture.slang:136]: This is not true. I need to delete this line.

File: tests/wgsl/texture.slang [hunk 4]
```diff
@@ -0,0 +1,451 @@
...
+
+        // Clamp variant
+        // WGSL doesn't support clamp variants for `textureSample()`
+
+        // ===============
+        // T SampleBias()
+        // https://www.w3.org/TR/WGSL/#texturesamplebias
+        // ===============
+
+        // WGSL doesn't support Bias for 1D texture
```
Comment (jkwak-work) [inline tests/wgsl/texture.slang:235]: I need to reflect this fact as `static_assert`
I will make some changes for that.

Comment (jkwak-work) [reply tests/wgsl/texture.slang:231]: I will do this in a separate PR later.

**Pull request merged**